lib/macho/macho_file.rb

```diff
@@ -264,6 +264,45 @@ def change_rpath(old_path, new_path)
       set_path_in_rpath(rpath_cmd, old_path, new_path)
     end
 
+    # Add the given runtime path to the Mach-O.
+    # @example
+    #  file.rpaths # => ["/lib"]
+    #  file.add_rpath("/usr/lib")
+    #  file.rpaths # => ["/lib", "/usr/lib"]
+    # @param path [String] the new runtime path
+    # @return [void]
+    # @raise [MachO::RpathExistsError] if the runtime path already exists
+    def add_rpath(path)
+      raise UnimplementedError.new("addition of rpaths")
+      raise RpathExistsError.new(path) if rpaths.include?(path)
+    end
+
+    # Delete the given runtime path from the Mach-O.
+    # @example
+    #  file.rpaths # => ["/lib"]
+    #  file.delete_rpath("/lib")
+    #  file.rpaths # => []
+    # @param path [String] the runtime path to delete
+    # @return void
+    # @raise [MachO::RpathUnknownError] if no such runtime path exists
+    def delete_rpath(path)
+      rpath_cmd = command(:LC_RPATH).find { |r| r.path.to_s == path }
```

Comment:
Hmm, this is a tricky one. I think the most "correct" behavior would be to delete all matching rpaths, although this is apparently not what `install_name_tool` does. Similarly, I think it makes a good amount of sense to prevent the user from adding/changing rpaths such that a duplicate is created.

This shouldn't break anything and should in principle be more "correct" than the official implementation, in the sense that a user's intentional rpath deletion will have the intended effect.